packages/ui/src/components/Table/useTableLogic.tsx

```diff
@@ -0,0 +1,135 @@
+
+const getHeadRow = <T extends GenericRowStructure>(headCells: HeadRowCells<T>) =>
+    ({
+        id: HEAD_ROW_IDENTIFIER,
+        cells: headCells,
+    } as HeadRow<T>);
+
+const useTableLogic = <T extends GenericRowStructure>({
+    headCells,
+    rows,
+    currentPage,
+    defaultRowsPerPage,
+    handlePageChange,
+    handleRowsPerPageChange,
+    handleSortCellClick,
+    onAllRowsSelectionChange,
+    onRowSelectionChange,
+    rowsPerPage,
+    searchProps,
+    selectedRowIds,
+    selectionType,
+    makeSearchableRowContent,
+    sortColumn,
+    sortDirection,
+}: BaseTableProps<T>) => {
+    const {
+        currentPageRows: internalCurrentPageRows,
+        handleRowsPerPageChange: internalHandleRowsPerPageChange,
+        handleSortCellClick: internalHandleSortCellClick,
+        filteredRows: internalFilteredRows,
+        handleChangePage: internalHandleChangePage,
+        handleChangeSearchValue: internalHandleChangeSearchValue,
+        handleRowSelection: internalHandleRowSelection,
+        handleAllRowsSelection: internalHandleAllRowsSelection,
+        state: {
+            selectedRowIds: internalSelectedRowIds,
+            rowsPerPage: internalRowsPerPage,
+            sortByColumnId: internalSortColumn,
+            searchValue: internalSearchValue,
+            sortDirection: internalSortDirection,
+        },
+        page: internalPage,
+    } = useTable<T>({
+        makeSearchableRowContent,
+        headCells,
+        rows,
+        selectionType,
+        onAllRowsSelectionChange,
+        onRowSelectionChange,
+        defaultRowsPerPage,
+        searchProps,
+    });
+
+    const tableState = {
+        selectedRowIds: selectedRowIds ?? internalSelectedRowIds,
+        page: currentPage ?? internalPage,
+        rowsPerPage: rowsPerPage ?? internalRowsPerPage,
+        sortByColumnId: sortColumn ?? internalSortColumn,
+        searchValue: searchProps?.value ?? internalSearchValue,
+        sortDirection: sortDirection ?? internalSortDirection,
+        currentPageRows: handleSortCellClick ? rows : internalCurrentPageRows,
+        filteredRows: handleSortCellClick ? rows : internalFilteredRows,
+        selectedRowsCount: getSelectedRowsCount(selectedRowIds ?? internalSelectedRowIds),
+        headRow: getHeadRow(headCells),
+    };
+
+    const tableHandlers = {
+        handleChangeSearchValue: searchProps?.onChange ?? internalHandleChangeSearchValue,
+        handleSortCellClick: handleSortCellClick ?? internalHandleSortCellClick,
+        handleRowsPerPageChange: handleRowsPerPageChange ?? internalHandleRowsPerPageChange,
+        handleChangePage: handlePageChange ?? internalHandleChangePage,
+        handleRowSelection: onRowSelectionChange ?? internalHandleRowSelection,
+        handleAllRowsSelection: onAllRowsSelectionChange ?? internalHandleAllRowsSelection,
+        // handled for both internal and external use cases
```

Comment:
not sure if this is needed since it doesn't change how the tableHandlers object is handled in the future. i.e. all handlers exposed will go here